pyacq/core/sharedarray.py

```diff
@@ -55,13 +43,62 @@ def __init__(self, shape=(1,), dtype = 'float64', shm_id = None):
                 self.mmap = mmap.mmap(self._tmpFile.fileno(), self.nbytes, mmap.MAP_SHARED, mmap.PROT_READ)
                 
     def close(self):
+        """Close this buffer.
+        """
         self.mmap.close()
         if not sys.platform.startswith('win') and hasattr(self, '_tmpFile'):
             self._tmpFile.close()
     
     def to_dict(self):
-        return {'shape': self.shape, 'dtype': self.dtype, 'shm_id': self.shm_id}
+        """Return a dict that can be serialized and sent to other processes to
+        access this buffer.
+        """
+        return {'nbytes': self.nbytes, 'shm_id': self.shm_id}
+    
+    def to_numpy(self, offset, dtype, shape, strides):
+        """Return a numpy array pointing to part (or all) of this buffer.
+        """
+        return np.ndarray(buffer=self.mmap, shape=shape,
+                          strides=strides, offset=offset, dtype=dtype)        
+        
+
+
```

Comment:
I have been thinking a lot about this. Let me see if I understand: The default is to have timeaxis=0 (sample, channel), which is trivial to write into a ring buffer. Right now using timeaxis=1 (channel, sample) does not make sense in RingBuffer because the data does not stack nicely along axis 0. In this case, are you suggesting splitting up the chunk as it is written into the buffer, such that each channel is written into a different part of the buffer to ensure they remain contiguous? For example, let's say the data is 4 samples by 2 channels:

    [1, 1, 1, 1, 2, 2, 2, 2]

If we write this into a ring buffer of length 8, then it should look like:
 
    [1, 1, 1, 1, 0, 0, 0, 0, 2, 2, 2, 2, 0, 0, 0, 0]

Is that correct? 
Why didn't I understand this sooner? 
I might not need the 1D ring buffer at all...